### ISSUE TITLE
Failure to recognize class array with PHP in @var

### ISSUE THREAD

**Issue by mikeryan776** [doxygen/doxygen#6831]
```
$ doxygen --version
1.8.15
```

The following declaration

```php
class Record implements DataRecordInterface
{

    /**
     * Array of data properties, keyed by property name.
     *
     * @var DataPropertyInterface[] $data
     */
    protected $data = [];
```

Produces 
```
.../Record.php:14: warning: documented symbol `Soong\Data\Record::DataPropertyInterface[] $data' was not declared or defined.
```

I've tried it with fully-qualifying DataPropertyInterface, and with `@property` instead with the same results. Elsewhere in the same project, `@return ClassName[]` works fine. I set up an `@param` which also worked. Using just DataPropertyInterface (whether or not fully-qualified) in @var works, so it is able to find the interface definition. The problem appears isolated to @var/@property with class arrays.

**Comment by albert-github** [comment]
Why do you want to use @var, the documentation is right in front of the variable so just remove the `@var DataPropertyInterface[] $data`.
Does this solve your problem or did I miss something (in the later case Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?)

**Comment by mikeryan776** [comment]
Well, I'm used to using @var since that's the convention in Symfony and Drupal. Also, the effect is different, without @var the type line is just treated as a description, with @var it is recognized as the member's type. For an example without the class array, with no `@var`:

```php
    /**
     * The value being wrapped.
     *
     * mixed $value
     */
    protected $value;
```

![novar](https://user-images.githubusercontent.com/616491/52502349-2a8e9c00-2ba8-11e9-8041-485653f77dfa.png)

And with the @var:

```php
    /**
     * The value being wrapped.
     *
     * @var mixed $value
     */
    protected $value;
```
![var](https://user-images.githubusercontent.com/616491/52502503-8f49f680-2ba8-11e9-9661-41d9eef72480.png)

Heading out on the road, if I can't throw together a standalone example in 10 minutes it'll have to wait a day or two...

Thanks.

**Comment by mikeryan776** [comment]
```
docs$ doxygen
.../dtest/src/Data/Record.php:14: warning: documented symbol `Soong\Data\Record::DataPropertyInterface[] $data' was not declared or defined.
.../dtest/src/Data/Record.php:17: warning: Member $data (variable) of class Soong::Data::Record is not documented.
```
[dtest.zip](https://github.com/doxygen/doxygen/files/2846677/dtest.zip)

**Comment by albert-github** [comment]
Looks like the `[]` is the problematic part.

**Comment by mikeryan776** [comment]
With

`@var DataPropertyInterface $data[]`

Doxygen renders

`DataPropertyInterface Soong\Data\Record::$data[] = []`

which seems less clear - one expects the type being returned, which is an array of DataPropertyInterface, to appear before the variable name. Still, it seems the best work-around for the moment.

The `DataPropertyInterface[]` syntax is used with `@var` in (at least) Symfony and Drupal, and it works fine with `@param` and `@return` - it seems reasonable to expect it to work here.

http://www.doxygen.nl/manual/commands.html#cmdvar says the "syntax is similar as for the phpDocumentor". http://docs.phpdoc.org/guides/types.html#arrays shows as the relevant example

`/** @var \DateTime[] An array of DateTime objects. */`

**Comment by albert-github** [comment]
The phpDocumentor documentation is a bit outdated (and links are broken) but the example shown signals that it should work like `DataPropertyInterface[]` and doxygen refers to it, also for the `@param` command.

Regarding the `@param` and `@return`, when I understand you correctly the `DataPropertyInterface[]`  with these commands (or did I misinterpret your statement):
- with the `@param` I get similar problems, about arguments not found in the argument list. Can you give an example
- with `@return` it looks like it is working but the first argument is taken and shown here. No check on the first word is done (so it looks like it is working, just try as return `0[]` o `boolean[]`)

**Comment by mikeryan776** [comment]
Ah, I tried again and I do see I get the errors for `@param` as well as `@var` - the resulting documentation looks right, which had me fooled, but the interface name in the param doc is not linked. So, it's not actually recognizing the type, it's just parroting what's on the `@param` line.

Thanks.

[dtest.zip](https://github.com/doxygen/doxygen/files/2851691/dtest.zip)

**Comment by albert-github** [comment]
@doxygen should things like `DataPropertyInterface[]` be recognized as data type in php code i.e. data type `DataPropertyInterface` abd append to the datat type `[]` ?

**Comment by doxygen** [comment]
@albert-github I think it makes sense to support [] after the type name to indicate an array of that type.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6841

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
